src/WebJobs.Script.WebHost/Management/InstanceManager.cs

```diff
@@ -80,21 +84,47 @@ public async Task<string> ValidateContext(HostAssignmentContext assignmentContex
         {
             _logger.LogInformation($"Validating host assignment context (SiteId: {assignmentContext.SiteId}, SiteName: '{assignmentContext.SiteName}')");
 
+            string error = null;
+
+            try
+            {
+                await ValidateContextInternal(assignmentContext);
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e.ToString());
+                error = "Invalid zip url specified";
+            }
+
+            return error;
+        }
+
+        public async Task ValidateContextInternal(HostAssignmentContext assignmentContext)
+        {
             var zipUrl = assignmentContext.ZipUrl;
+            HttpResponseMessage response = null;
             if (!string.IsNullOrEmpty(zipUrl))
             {
                 // make sure the zip uri is valid and accessible
-                var request = new HttpRequestMessage(HttpMethod.Head, zipUrl);
-                var response = await _client.SendAsync(request);
-                if (!response.IsSuccessStatusCode)
+                await Utility.InvokeWithRetriesAsync(async () =>
                 {
-                    string error = $"Invalid zip url specified (StatusCode: {response.StatusCode})";
-                    _logger.LogError(error);
-                    return error;
-                }
+                    try
+                    {
+                        using (_metricsLogger.LatencyEvent(MetricEventNames.LinuxContainerSpecializationZipHead))
+                        {
+                            var request = new HttpRequestMessage(HttpMethod.Head, zipUrl);
+                            response = await _client.SendAsync(request);
+                            response.EnsureSuccessStatusCode();
+                        }
+                    }
+                    catch (Exception e)
+                    {
+                        string error = $"Invalid zip url specified (StatusCode: {response?.StatusCode})";
+                        _logger.LogError(error, e);
```

Comment:
This is the wrong overload - take a look at the method definition. You want LogError(e, error)

Reply:
Fixed